### PR TITLE
Default button actions overrides

### PR DESCRIPTION
**Pull request by vicfergar**
This PR includes a new configuration option called _start_action_ that can be used to define which service to call when the start button is pressed.

I have updated documentation file too.

This can solve issue #158

### CODE REVIEW

File: src/vacuum-card.js [hunk 1]
```diff
@@ -168,6 +168,19 @@ class VacuumCard extends LitElement {
...
+	if (!start_action) {
+		this.callService('start');
+		return;
+	}
+	
+	var service = start_action.service;
+	var service_data = start_action.service_data;
+	const [domain, name] = service.split('.');
+    this.hass.callService(domain, name, service_data);
+  }
```
Comment (denysdovhan) [inline src/vacuum-card.js:183]:
```suggestion
  handleStart() {
    if (!this.config.start_action) {
      this.callService('start');
      return;
    }
	
    const {service, service_data} = this.config.start_action;
    const [domain, name] = service.split('.');
    this.hass.callService(domain, name, service_data);
  }
```

File: src/vacuum-card.js [hunk 2]
```diff
@@ -348,7 +361,7 @@ class VacuumCard extends LitElement {
       case 'paused': {
         return html`
           <div class="toolbar">
-            <paper-button @click="${() => this.callService('start')}">
+            <paper-button @click="${() => this.handleStart()}">
```
Comment (denysdovhan) [inline src/vacuum-card.js:364]:
```suggestion
            <paper-button @click="${this.handleStart}">
```

File: src/vacuum-card.js [hunk 3]
```diff
@@ -363,7 +376,7 @@ class VacuumCard extends LitElement {
       case 'returning': {
         return html`
           <div class="toolbar">
-            <paper-button @click="${() => this.callService('start')}">
+            <paper-button @click="${() => this.handleStart()}">
```
Comment (denysdovhan) [inline src/vacuum-card.js:379]:
```suggestion
            <paper-button @click="${this.handleStart}">
```

File: src/vacuum-card.js [hunk 4]
```diff
@@ -405,7 +418,7 @@ class VacuumCard extends LitElement {
             <ha-icon-button
               icon="hass:play"
               title="${localize('common.start')}"
-              @click="${() => this.callService('start')}"
+              @click="${() => this.handleStart()}"
```
Comment (denysdovhan) [inline src/vacuum-card.js:421]:
```suggestion
              @click="${this.handleStart}"
```

**Review by denysdovhan — CHANGES_REQUESTED** [review]
I've suggested a few changes. Please, test locally.

**Review by vicfergar — COMMENTED** [review]
Proposed changes included and tested locally. Thanks!

**Comment by vicfergar** [comment]
There is something that maybe should be changed. Differentiate between _HandleStart_ and _HandleResume_, because probably for most cases only _HandleStart_ must be replaced.

Would you like if I add this distinction and include custom action (resume_action) just in case someone needs to override the default behavior for resume too?

**Comment by denysdovhan** [comment]
@vicfergar yes, I'd like to. Maybe we can also add custom handlers for stop and pause. Grouping them under the prop would be great. 

I'd like to call it `actions` and rename `actions` to `shortcuts`, but it's gonna be a breaking change 😞

**Comment by vicfergar** [comment]
@denysdovhan I'm agree with that. If breaking change is an issue (I suppose it can be 😄), what about a group called _custom_actions_ or _override_actions_ and rename current _actions_ to _shortcuts_ (while accepting actions as backward compatibility)

Something like:
```
type: 'custom:vacuum-card'
entity: vacuum.vacuum_cleaner
override_actions:
  start:
    service: xiaomi_miio.vacuum_clean_segment
    service_data:
      entity_id: vacuum.vacuum_cleaner
      segments: [16, 20]
  resume:
    service: script.custom_resume
  pause:
    service: script.custom_pause
  stop:
    service: script.custom_stop
  return_to_base:
    service: script.custom_return_to_base
  localize:
    service: script.custom_localize
shortcuts: #or actions for backward compatibility
  - name: Clean living room
    service: script.clean_living_room
    icon: 'mdi:sofa'
```

What do you think?

**Comment by denysdovhan** [comment]
We can release a breaking version — this is not a problem. It's relatively easy to check what we have in `actions`. If it's an object — it's an override for `start`, `stop`, etc; if it's an array — it's a link of shortcuts and we should display a deprecation warning and do not render them.

**Comment by vicfergar** [comment]
Hi @denysdovhan I tried to acomplish the proposed solution.

Please let me know if you would like to change something or propose any change if needed 😉

**Comment by vicfergar** [comment]
Hi @denysdovhan 

Any news on this? I have been using this branch without problems for a while since our last conversation.

File: README.md [hunk 5]
```diff
@@ -58,13 +58,18 @@ This card can be configured using Lovelace UI editor.
...
 
-_Sorry, no support for `actions` and `stats` in visual config yet._
+_Sorry, no support for `start_action`, `actions` and `stats` in visual config yet._
 
 Typical example of using this card in YAML config would look like this:
 
 ```yaml
 type: 'custom:vacuum-card'
 entity: vacuum.vacuum_cleaner
+start_action:
```
Comment (denysdovhan) [inline README.md:68]: Please update cover `shortcuts` and `actions` in documentation

**Review by denysdovhan — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (vicfergar) [reply README.md]: I have updated documentation, let me know if it correct for you

Comment (vicfergar) [reply README.md]: Hi @denysdovhan, can we resolve this?

**Comment by Mikhail5555** [comment]
If its not too big of a feature creep, could you maybe add the option to completely disable the default buttons? (I personally use only selected rooms cleaning which renders that button useless and only prone to misclicks)

**Review by denysdovhan — APPROVED** [review]

**Comment by denysdovhan** [comment]
Thanks for contributing!

**Pull request merged**